### PR TITLE
867 modify mapping matrix

### PR DESCRIPTION
**Pull request by chlebowa**
Related to [this issue](https://github.com/insightsengineering/teal/issues/867).

`FilteredData` receives a new public method, `$get_available_teal_slices` that returns the contents of the private field `private$available_teal_slices`.

### CODE REVIEW

File: R/FilteredData.R [hunk 1]
```diff
@@ -155,6 +155,16 @@ FilteredData <- R6::R6Class( # nolint
       invisible(NULL)
     },
 
+    #' Get list of filter states available for this object.
```
Comment (gogonzo) [inline R/FilteredData.R:158]: https://github.com/insightsengineering/teal/blob/867_mapping_matrix_NA%40main/R/module_filter_manager.R#L208
`set_available_teal_slices` filters only available datanames - but should it also filter by `attr(self$get_filter_state(), "include_varnames")`?

Comment (chlebowa) [reply R/FilteredData.R:158]: Good question. 
`FilterStates` will not allow such a state to be created from the UI but what if a `teal_slices` is passed that refers to forbidden variables? This is why #340 would be nice to have.

Let me run some tests. I will modify `$set_available_teal_slices` if necessary.

Comment (gogonzo) [reply R/FilteredData.R:158]: Yup, #340 is nice to have but MAE make `teal_slice` objects hard to match with include/exclude_varnames without introducing MAE-specific-code (against open-close principle).

Comment (chlebowa) [reply R/FilteredData.R:174]: `teal_slice_expr` will be included if `dataname` checks out, even if they refer to variables not allowed.

File: R/FilteredData.R [hunk 2]
```diff
@@ -150,11 +152,35 @@ FilteredData <- R6::R6Class( # nolint
       checkmate::assert_class(x, "reactive")
       private$available_teal_slices <- reactive({
         # Available filters should be limited to the ones relevant for this FilteredData.
-        Filter(function(x) x$dataname %in% self$datanames(), x())
+        allowed <- attr(self$get_filter_state(), "include_varnames")
+        forbidden <- attr(self$get_filter_state(), "exclude_varnames")
```
Comment (gogonzo) [inline R/FilteredData.R:156]: We want this reactive to be dependent on `x` (slices_global)

```suggestion
        allowed <- attr(isolate(self$get_filter_state()), "include_varnames")
        forbidden <- attr(isolate(self$get_filter_state()), "exclude_varnames")
```

**Review by gogonzo — APPROVED** [review]
👍

**Pull request merged**